spec/api/students.spec.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Created by rauldablaing on 1/19/15.
+ */
+
+/* jshint quotmark:false */
+
+// Add dependencies
+var frisby = require('frisby');
+// Set up variables for info you will use often
+var url = 'http://localhost:3000/api/students/';
+var initialRecord = {
+  "firstName": "Jim",
+  "lastName": "Smith",
+  "courses": "JavaScript"
```

Comment:
There's no courses property when you post to api/students/ which is why you get the error below.
<pre><code>Failures:

  1) Frisby Test: Create a contact with post
        [ POST http://localhost:3000/api/students/ ]
   Message:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
   Stacktrace:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1271:17)
    at jasmine.Matchers.toContainJson (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1186:12)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:717:24)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1074:43)
    at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)

  2) Frisby Test: Enforce mandatory fields when creating
        [ POST http://localhost:3000/api/students/ ]
   Message:
     Error: Expected 'undefined' to be type 'array' for comparison
   Stacktrace:
     Error: Expected 'undefined' to be type 'array' for comparison
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1238:11)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at jasmine.Matchers.toContainJson (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1186:12)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:717:24)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1074:43)

  3) Frisby Test: Get contact using id
        [ GET http://localhost:3000/api/students/54c29132024ef9f80e843e01 ]
   Message:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
   Stacktrace:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1271:17)
    at jasmine.Matchers.toContainJson (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1186:12)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:717:24)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1074:43)
    at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)

Finished in 0.135 seconds
4 tests, 15 assertions, 3 failures, 0 skipped</code></pre>